.github/workflows/site-build.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and Deploy LoopBack.io site
+
+on:
+  push:
+    branches:
+      - gh-pages
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+        with:
+          persist-credentials: false
+
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+
+      - name: Set up Ruby 2.7
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+
+      - name: Install and Build 🔧
+        run: |
+          bundle install
+          npm ci
+          npm ls @loopback/docs 2>/dev/null || true
+          npm run build
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@3.6.2
```

Comment:
Do we need the deploy step? Isn't GitHub already deploying changes automatically for us?

Reply:
See https://github.com/strongloop/loopback.io/settings

<img width="921" alt="Screen Shot 2021-01-08 at 10 42 07" src="https://user-images.githubusercontent.com/1140553/104000193-2c02dd80-519e-11eb-929a-6cf3ba8b3ba9.png">

Reply:
BTW I found it always annoying that this repo is using `gh-pages` as the main branch. It would be great to eventually change the configuration to use `main` (or `master`) instead. Such effort is out of scope of this pull request, I am just mentioning it here because it's related to the deployment process.

Reply:
IIRC, `gh-pages` is the default branch for github pages.

Reply:
> Do we need the deploy step? Isn't GitHub already deploying changes automatically for us?

Reply:
> IIRC, `gh-pages` is the default branch for github pages.

Yes, it's the default, but the value can be changed. In the past, GitHub mandated `gh-pages` names for certain types of repositories, but that restriction was removed some time ago.